src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -1307,17 +1307,57 @@ public IEnumerable<UnspentOutputReference> GetSpendableTransactionsInAccount(Wal
             Wallet hdWallet = this.GetWallet(walletAccountReference.WalletName);
             HdAccount hdAccount = this.GetAccounts(hdWallet, walletAccountReference.AccountName).FirstOrDefault();
 
-            var extPubKey = ExtPubKey.Parse(hdAccount.ExtendedPubKey, this.Network);
-
             var spendable = conn.GetSpendableOutputs(walletContainer.Wallet.WalletId, hdAccount.Index, currentChainHeight, coinBaseMaturity ?? this.Network.Consensus.CoinbaseMaturity, confirmations);
 
+            ExtPubKey extPubKey = null;
+
+            // The extPubKey will be null if this is a watch only account.
+            if (hdAccount.ExtendedPubKey != null)
+            {
+                extPubKey = ExtPubKey.Parse(hdAccount.ExtendedPubKey, this.Network);
+            }
+
             var cachedPubKeys = new Dictionary<KeyPath, PubKey>();
 
             foreach (HDTransactionData transactionData in spendable)
             {
+                int tdConfirmations = (transactionData.OutputBlockHeight == null) ? 0 : (currentChainHeight + 1) - (int)transactionData.OutputBlockHeight;
+
+                if (extPubKey == null)
+                {
+                    // We cannot derive a pubKey from the scriptPubKey as it has been hashed already.
+                    Script watchOnlyScriptPubKey = Script.FromHex(transactionData.ScriptPubKey);
+
+                    HdAddress watchOnlyAddress = this.ToHdAddress(new HDAddress()
+                    {
+                        AccountIndex = transactionData.AccountIndex,
+                        AddressIndex = transactionData.AddressIndex,
+                        AddressType = (int)transactionData.AddressType,
+                        PubKey = watchOnlyScriptPubKey.ToHex(),
+                        ScriptPubKey = watchOnlyScriptPubKey.ToHex(),
+                        Address = transactionData.Address,
+                        // TODO: It may be possible to construct a P2WPKH address using the P2PKH scriptPubKey as a starting point
```

Comment:
Each HDTransactionData is linked to a HDAddress which contains the Bech32Address.